libraries/chain/controller.cpp

```diff
@@ -889,7 +897,7 @@ struct controller_impl {
                   section.read_row(legacy_global_properties, db);
 
                   db.create<global_property_object>([&](auto& gpo ){
-                     gpo.initalize_from(legacy_global_properties);
+                     gpo.initalize_from(legacy_global_properties, chain_id);
```

Comment:
Along with this I would either need to make chain_id an fc::optional or else compare it to chain_id_type() to validate the command line parameter here (rather than in chain_plugin), which of those 2 is your preference? I'm presuming fc::optional.

Reply:
The code in controller should not validate anything against command line parameters. It should just assume any sanity checks have already been done in chain_plugin, and just calculate the chain ID from the genesis state in these legacy snapshots.